src/rust/engine/fs/store/src/lib.rs

```diff
@@ -597,6 +600,24 @@ impl Store {
       .to_boxed()
   }
 
+  ///
+  /// Ensure that a file is locally loadable, which will download it from the Remote store as
+  /// a sideeffect (if one is configured). Called only with the Digest of a File.
+  ///
+  pub async fn ensure_local_has_file(&self, file_digest: Digest) -> Result<(), String> {
+    let result = self
+      .load_bytes_with(EntryType::File, file_digest, |_| Ok(()), |_| Ok(()))
+      .await?;
+    if result.is_some() {
```

Comment:
Style-nit: maybe do a `match` here since you are deconstructing `result` any way? It would at least ensure statically that all branches match exhaustively.

```
match result {
  Some(_) => Ok(()),
  _ => Err(...),
}
```